src/character.cpp

```diff
@@ -9184,6 +9185,14 @@ void Character::cancel_activity()
     if( has_activity( ACT_TRY_SLEEP ) ) {
         remove_value( "sleep_query" );
     }
+    if( has_activity( ACT_LOCKPICK ) ) {
+        std::vector<item *> bio_picklocks = g->u.items_with( []( const item & itm ) {
+            return itm.typeId() == "pseudo_bio_picklock";
+        } );
+        if( !bio_picklocks.empty() ) {
+            g->u.i_rem( bio_picklocks[0] );
+        }
+    }
```

Comment:
Hmm, this still seems a little fragile as this wont be run if the activity is stopped using `player_activity::set_to_null()` for example. I think we could instead set a variable on the activity to indicate that it is using the bionic lockpick. Then we could insert the pseudo item into the player's inventory on completion of the activity if that variable is set. What do you think?

Reply:
>if the activity is stopped using player_activity::set_to_null()

I didn't find any cases which could stop this activity by the means of `set_to_null`. Could you please list any?

> Then we could insert the pseudo item into the player's inventory on completion of the activity if that variable is set.

The item must be inserted prior to completion of activity, not after it.

Reply:
> I didn't find any cases which could stop this activity by the means of set_to_null. Could you please list any?

It may very well be fine with the current code. When i said "fragile" i meant likely to break and cause bugs in the future if the code is changed. 

> The item must be inserted prior to completion of activity, not after it.

This behavior isn't set in stone, it could be changed. One way to do this would be to set the first member of `player_activity::str_values` to `bio_picklock` if using the fingerpick and check for that string before checking the `targets` vector at the start of the finish function.  If that string is present, the pseudo item could then be inserted into the players inventory and used instead of the first `item_location` in the `targets` vector. In my opinion, such an approach would be less prone to cause problems in the future.